aws/src/main/java/org/apache/iceberg/aws/dynamodb/DynamoDbLockManager.java

```diff
@@ -0,0 +1,374 @@
+    boolean succeeded = false;
+    DynamoHeartbeat heartbeat = heartbeats.get(entityId);
+    try {
+      Tasks.foreach(entityId)
+          .retry(RELEASE_RETRY_ATTEMPTS_MAX)
+          .throwFailureWhenFinished()
+          .onlyRetryOn(
+              ProvisionedThroughputExceededException.class,
+              TransactionConflictException.class,
+              RequestLimitExceededException.class,
+              InternalServerErrorException.class)
+          .run(id -> dynamo.deleteItem(DeleteItemRequest.builder()
+              .tableName(lockTableName)
+              .key(toKey(id))
+              .conditionExpression(CONDITION_LOCK_ID_MATCH)
+              .expressionAttributeValues(toLockIdValues(id, ownerId))
+              .build()));
+      succeeded = true;
+    } catch (ConditionalCheckFailedException e) {
+      LOG.error("Failed to release lock for entity: {}, owner: {}, lock entity does not exist or owner not match",
+          entityId, ownerId, e);
+    } catch (DynamoDbException e) {
+      LOG.error("Failed to release lock {} by for entity: {}, owner: {}, encountered unexpected DynamoDB exception",
+          entityId, ownerId, e);
+    } finally {
+      if (heartbeat != null && heartbeat.ownerId().equals(ownerId)) {
+        heartbeat.cancel();
+      }
+    }
+
+    return succeeded;
+  }
+
+  private static Map<String, AttributeValue> toKey(String entityId) {
+    return ImmutableMap.of(COL_LOCK_ENTITY_ID, AttributeValue.builder().s(entityId).build());
+  }
+
+  private static Map<String, AttributeValue> toNewItem(String entityId, String ownerId, long heartbeatTimeoutMs) {
+    return ImmutableMap.of(
+        COL_LOCK_ENTITY_ID, AttributeValue.builder().s(entityId).build(),
+        COL_LOCK_OWNER_ID, AttributeValue.builder().s(ownerId).build(),
+        COL_VERSION, AttributeValue.builder().s(UUID.randomUUID().toString()).build(),
+        COL_LEASE_DURATION_MS, AttributeValue.builder().n(Long.toString(heartbeatTimeoutMs)).build());
+  }
+
+  private static Map<String, AttributeValue> toLockIdValues(String entityId, String ownerId) {
+    return ImmutableMap.of(
+        ":eid", AttributeValue.builder().s(entityId).build(),
+        ":oid", AttributeValue.builder().s(ownerId).build());
+  }
+
+  @Override
+  public void close() {
+    dynamo.close();
+    heartbeats.values().forEach(DynamoHeartbeat::cancel);
+    heartbeats.clear();
+  }
+
+  /**
+   * The lock table schema, for users who would like to create the table separately
+   * @return lock table schema
+   */
+  public static List<KeySchemaElement> lockTableSchema() {
+    return LOCK_TABLE_SCHEMA;
+  }
+
+  /**
+   * The lock table column definition, for users who whould like to create the table separately
+   * @return lock table column definition
+   */
+  public static List<AttributeDefinition> lockTableColDefinitions() {
+    return LOCK_TABLE_COL_DEFINITIONS;
+  }
+
+  private static class DynamoHeartbeat implements Runnable {
```

Comment:
[minor] should we also make this `DynamoDbHeartbeat` following `Dynamo` -> `DynamoDb` convention

Reply:
updated